go/cmd/goserver/server.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"context"
+	"github.com/feast-dev/feast/go/internal/feast"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+)
+
+type servingServiceServer struct {
+	fs *feast.FeatureStore
+	serving.UnimplementedServingServiceServer
+}
+
+func (s *servingServiceServer) GetFeastServingInfo(ctx context.Context, request *serving.GetFeastServingInfoRequest) (*serving.GetFeastServingInfoResponse, error) {
+	return &serving.GetFeastServingInfoResponse{
+		Version: feastServerVersion,
+	}, nil
+}
+
+func (s *servingServiceServer) GetOnlineFeatures(ctx context.Context, request *serving.GetOnlineFeaturesRequest) (*serving.GetOnlineFeaturesResponse, error) {
+	s.fs.SetOnlineStoreContext(ctx)
+	return s.fs.GetOnlineFeatures(request)
+}
```

Comment:
If I'm reading this correctly there's a race condition, right? Because concurrent requests to `GetOnlineFeatures` may set the `ctx` in the Online store? Why not just pass the ctx into the `fs.GetOnlineFeatures` method directly?

Reply:
can we not do this set context at all and instead pass it everywhere as a parameter, as it being done in all go libraries.